qutip/models/tests/test_dicke.py

```diff
@@ -0,0 +1,309 @@
+
+    def test_jmm1_dictionary(self):
+        """
+        Test the function to generate the mapping from jmm1 to ik matrix
+        """
+        d1, d2, d3, d4 = jmm1_dictionary(1)
+
+        d1_correct = {(0, 0): (0.5, 0.5, 0.5), (0, 1): (0.5, 0.5, -0.5),
+                        (1, 0): (0.5, -0.5, 0.5), (1, 1): (0.5, -0.5, -0.5)}
+
+        d2_correct = {(0.5, -0.5, -0.5): (1, 1),(0.5, -0.5, 0.5): (1, 0),
+                            (0.5, 0.5, -0.5): (0, 1),
+                            (0.5, 0.5, 0.5): (0, 0)}
+
+        d3_correct = {0: (0.5, 0.5, 0.5), 1: (0.5, 0.5, -0.5),
+                        2: (0.5, -0.5, 0.5),
+                        3: (0.5, -0.5, -0.5)}
+
+        d4_correct = {(0.5, -0.5, -0.5): 3, (0.5, -0.5, 0.5): 2, 
+                        (0.5, 0.5, -0.5): 1, (0.5, 0.5, 0.5): 0}
+
+        assert_equal(d1, d1_correct)
+        assert_equal(d2, d2_correct)
+        assert_equal(d3, d3_correct)
+        assert_equal(d4, d4_correct)
+
+
+        d1, d2, d3, d4 = jmm1_dictionary(2)
+
+        d1_correct = {(3, 3): (0.0, -0.0, -0.0), (2, 2): (1.0, -1.0, -1.0),
+                        (2, 1): (1.0, -1.0, 0.0), (2, 0): (1.0, -1.0, 1.0),
+                        (1, 2): (1.0, 0.0, -1.0), (1, 1): (1.0, 0.0, 0.0),
+                        (1, 0): (1.0, 0.0, 1.0), (0, 2): (1.0, 1.0, -1.0),
+                        (0, 1): (1.0, 1.0, 0.0), (0, 0): (1.0, 1.0, 1.0)}
+
+        d2_correct = {(0.0, -0.0, -0.0): (3, 3), (1.0, -1.0, -1.0): (2, 2),
+                        (1.0, -1.0, 0.0): (2, 1), (1.0, -1.0, 1.0): (2, 0),
+                        (1.0, 0.0, -1.0): (1, 2), (1.0, 0.0, 0.0): (1, 1),
+                        (1.0, 0.0, 1.0): (1, 0), (1.0, 1.0, -1.0): (0, 2),
+                        (1.0, 1.0, 0.0): (0, 1), (1.0, 1.0, 1.0): (0, 0)}
+
+        d3_correct ={15: (0.0, -0.0, -0.0), 10: (1.0, -1.0, -1.0),
+                        9: (1.0, -1.0, 0.0), 8: (1.0, -1.0, 1.0),
+                        6: (1.0, 0.0, -1.0), 5: (1.0, 0.0, 0.0),
+                        4: (1.0, 0.0, 1.0), 2: (1.0, 1.0, -1.0),
+                        1: (1.0, 1.0, 0.0), 0: (1.0, 1.0, 1.0)}
+
+        d4_correct = {(0.0, -0.0, -0.0): 15, (1.0, -1.0, -1.0): 10,
+                        (1.0, -1.0, 0.0): 9, (1.0, -1.0, 1.0): 8,
+                        (1.0, 0.0, -1.0): 6, (1.0, 0.0, 0.0): 5,
+                        (1.0, 0.0, 1.0): 4, (1.0, 1.0, -1.0): 2,
+                        (1.0, 1.0, 0.0): 1, (1.0, 1.0, 1.0): 0}
+        
+        assert_equal(d1, d1_correct)
+        assert_equal(d2, d2_correct)
+        assert_equal(d3, d3_correct)
+        assert_equal(d4, d4_correct)
+    
+    def test_lindbladian(self):
+        """
+        Test the generation of the Lindbladian matrix
+        """
+        N = 1
+        gCE = 0.5
+        gCD = 0.5
+        gCP = 0.5
+        gE = 0.1
+        gD = 0.1
+        gP = 0.1
+
+        system = Piqs(N = N, emission = gE, pumping = gP, dephasing = gD,
```

Comment:
spaces in kwargs. Quite a few other examples too, which I will not mark all of them